functions/index.js

```diff
@@ -1,7 +1,54 @@
 const functions = require('firebase-functions')
 const express = require('express')
 const app = express()
-app.get('/', (req, res) => {
-  res.send('hello world!')
+const bodyParser = require('body-parser')
+const admin = require('firebase-admin')
+const cors = require('cors')
+
+admin.initializeApp(functions.config().firebase);
+app.use(cors({ origin: true }));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+const db = admin.firestore();
+const requestCollectionName = 'tutoringRequests';
+
+app.post('/api/requests', async (req, res) => {
+  // functions.logger.log(req)
+  try {
+    const request = {
+      name: req.body.name,
+      pantherID: req.body.pantherID,
+      course: req.body.course,
+      completed: false,
+      requestedAt: new Date().toString()
+    }
+    const requestCollection = db.collection(requestCollectionName);
+    const querySnapshot = await requestCollection.where("pantherID", "==", request.pantherID).where("course", "==", request.course ).get()
```

Comment:
Put the query and the add calls to the db in separate try/catch blocks. That way if something goes wrong with one of the calls, we can have different logs for each call in the catch blocks.

Reply:
Believe that I changed this to what you intended. Please let me know if this is not the case.